### DISCUSSION TITLE
Can anyone share a backup script?

### DISCUSSION THREAD

**Discussion by justdaj** [mastodon/mastodon#22215] [Q&A]
I'm a bit of a novice at this, but learning fast and enjoying it.

My Mastodon server is up and running, and I've upgraded several versions to 4.0.2.  Each time I took a backup by imaging the disk, but that's not really practical going forward for regular backups.

I've read the documentation on what to backup, and I can do it manually but I need to be various users in order to get all the data.

I need to be mastodon user to pg_dump mastodon_production (I'd rather pg_dumpall but don't seem to have permissions for that as any user!).
I can grab the .env.production file and all the files from live/public/system

I have to be root to get /var/lib/redis/dump.rdb  :-(

I'd love one script that does it all, with all the correct permissions.

FYI -- my install is on Ubuntu and I followed the install on joinmastodon.org -- did not deviate in any way.

thanks for any support

**Comment by capcom1701** [comment]
I ask in ignorance, but is it possible to put the pg_dump command as a daily line in the crontab for the mastodon superuser and then the redis dump as a line in the root crontab?
